From 712993ed978e2e443fb327a7347393c81b79d4b9 Mon Sep 17 00:00:00 2001
From: Akhil Xavier <akhilxavier@inforcecomputing.com>
Date: Wed, 4 Jul 2018 16:48:10 +0530
Subject: [PATCH 7/7] device tree:Changes made for overlay support

A generic dt node kept for csi0 interface. Respective sensors needed to be enabled using overlay.
---
 arch/arm64/boot/dts/qcom/apq8096-db820c.dtsi | 38 +++-------------------------
 1 file changed, 4 insertions(+), 34 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/apq8096-db820c.dtsi b/arch/arm64/boot/dts/qcom/apq8096-db820c.dtsi
index 633859b..2031834 100644
--- a/arch/arm64/boot/dts/qcom/apq8096-db820c.dtsi
+++ b/arch/arm64/boot/dts/qcom/apq8096-db820c.dtsi
@@ -270,9 +270,8 @@
 			status = "ok";
 
 			camera_rear@3b {
-				compatible = "ovti,ov5640";
-				reg = <0x3b>;
-
+				/* compatible, reg, clock and status will be enabled using dt overlay */
+				/* Rest of hw parameters are same for the interface */
 				enable-gpios = <&msmgpio 29 GPIO_ACTIVE_HIGH>;
 				reset-gpios = <&msmgpio 30 GPIO_ACTIVE_LOW>;
 				pinctrl-names = "default";
@@ -280,7 +279,6 @@
 
 				clocks = <&mmcc CAMSS_MCLK0_CLK>;
 				clock-names = "xclk";
-				clock-frequency = <23880000>; //
 
 				vdddo-supply = <&camera_vdddo_1v8>;
 				vdda-supply = <&camera_vdda_2v8>;
@@ -289,7 +287,7 @@
 				status = "disabled";
 
 				port {
-					ov56401_ep: endpoint {
+					rear_ep: endpoint {
 						clock-lanes = <1>;
 						data-lanes = <0 2>;
 						remote-endpoint = <&csiphy0_ep>;
@@ -297,34 +295,6 @@
 				};
 			};
 
-                        camera_rear@10 {
-                                compatible = "sony,imx230";
-                                reg = <0x10>;
-
-                                enable-gpios = <&msmgpio 29 GPIO_ACTIVE_HIGH>;
-                                reset-gpios = <&msmgpio 30 GPIO_ACTIVE_LOW>;
-                                pinctrl-names = "default";
-                                pinctrl-0 = <&camera_rear_default>;
-
-                                clocks = <&mmcc CAMSS_MCLK0_CLK>;
-                                clock-names = "xclk";
-                                clock-frequency = <23880000>;
-
-                                vdddo-supply = <&camera_vdddo_1v8>;
-                                vdda-supply = <&camera_vdda_2v8>;
-                                vddd-supply = <&camera_vddd_1v5>;
-
-                                status = "disabled";
-
-                                port {
-                                        ov5640_ep: endpoint {
-                                                clock-lanes = <1>;
-                                                data-lanes = <0 2>;
-                                                remote-endpoint = <&csiphy0_ep>;
-                                        };
-                                };
-                        };
-
 			camera_board@39 {
 				compatible = "ovti,ov5645";
 				reg = <0x39>;
@@ -393,7 +363,7 @@
 					csiphy0_ep: endpoint {
 						clock-lanes = <7>;
 						data-lanes = <0 1>;
-						remote-endpoint = <&ov5640_ep>;
+						remote-endpoint = <&rear_ep>;
 						status = "ok";
 					};
 				};
-- 
2.7.4

